src/client.rs

```diff
@@ -142,12 +115,22 @@ pub trait ClientContext: Send + Sync {
         error!("librdkafka: {}: {}", error, reason);
     }
 
-    /// Generates the OAuth token.
+    /// Refreshes an OAuth token from the provided configuration.
+    ///
+    /// Override with an appropriate implementation when using the `OAUTHBEARER`
+    /// SASL authentication mechanism. For this method to be called, you must
+    /// also set [`ClientContext::ENABLE_REFRESH_OAUTH_TOKEN_CALLBACK`] to true.
+    ///
```

Comment:
I don't think that this description does as good job of telling the user what they need to actually do for their implementation of this function. If I saw that the function name was `refresh_oauth_token`, I'd assume that I have to call `rd_kafka_oauthbearer_set_token` or `rd_kafka_oauthbearer_set_token_failure` in my implementation, which is not what we want.

I think we should be explicit that the expectation for this function is _only_ that the user create the token and return it (or some compatible error).

Reply:
Fair enough. I backed out the rename.